x-pack/plugins/observability/public/pages/slos/components/slo_list.tsx

```diff
@@ -61,18 +63,33 @@ export function SloList({ autoRefresh }: Props) {
     storeState({ page: 0, sort: { by: newSort, direction: state.sort.direction } });
   };
 
+  const handleChangeViewMode = (newViewMode: ViewMode) => {
+    setViewMode(newViewMode);
+    storeState({ viewMode: newViewMode });
+  };
+
   return (
     <EuiFlexGroup direction="column" gutterSize="m" data-test-subj="sloList">
       <EuiFlexItem grow>
         <SloListSearchBar
           loading={isLoading || isCreatingSlo || isCloningSlo || isUpdatingSlo || isDeletingSlo}
           onChangeQuery={handleChangeQuery}
           onChangeSort={handleChangeSort}
+          onChangeViewMode={handleChangeViewMode}
           initialState={state}
         />
       </EuiFlexItem>
       <EuiFlexItem>
-        <SloListItems sloList={results} loading={isLoading || isRefetching} error={isError} />
+        {viewMode === 'compact' && (
```

Comment:
@kdelemme We could do this check one level up in this [file](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/public/pages/slos/slos.tsx#L97), no? There you have the logic for checking for proper license as well.

Reply:
The view mode toggle is in the SloListSearchBar used in the file, and this is common to all view mode, we only change the component related to the selected view mode. I'm not sure how this would work if lifted above?

Reply:
I see `SloListSearchBar` is in this file, so I guess it makes sense to keep it here. I just had in my mind `compactView`, `listView`, `cardView` and I associated `slo_list` file with the listView. I would assume view mode toggle was in the `slos` file. In that case it would make perfect sense to have the logic of changing the component according to the selected view mode in the slos file. But I see we have a different structure here.